tests/unit/test_date_compliance.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Tests for extension interface compliance, inherited from pandas.
+
+See:
+https://github.com/pandas-dev/pandas/blob/main/pandas/tests/extension/test_period.py
+"""
+
+import datetime
+
+import numpy
+from pandas.tests.extension import base
+import pytest
+
+from db_dtypes import DateArray
+
+# NDArrayBacked2DTests suite added in https://github.com/pandas-dev/pandas/pull/44974
+pytest.importorskip("pandas", minversion="1.5.0dev")
+
+
+@pytest.fixture
+def data():
+    return DateArray(
+        numpy.arange(
+            datetime.datetime(1900, 1, 1),
+            datetime.datetime(2099, 12, 31),
+            datetime.timedelta(days=13),
+            dtype="datetime64[ns]",
+        )
+    )
+
+
+@pytest.fixture
+def data_missing():
+    return DateArray([None, datetime.date(2022, 1, 27)])
+
+
+class Test2DCompat(base.NDArrayBacked2DTests):
```

Comment:
Note that this class is only one of the base test classes you can inherit from to test compliance (and actually a class that tests functionality (2D aspect of the EA) that is currently not used in pandas). 
You probably want to add the others as well (see https://github.com/pandas-dev/pandas/blob/main/pandas/tests/extension/base/__init__.py for the different classes, and https://github.com/pandas-dev/pandas/blob/main/pandas/tests/extension/decimal/test_decimal.py for an example that uses them)